lib/engine/company.rb

```diff
@@ -44,7 +44,9 @@ def min_bid
     def close!
       @closed = true
 
-      all_abilities.each { |a| remove_ability(a) }
+      while (a = all_abilities.first)
```

Comment:
change to all_abilities.dup.each

Reply:
Okay. I went with the while loop because I thought it would be more efficient than calling the constructor for each entry.